### PR TITLE
Display friendlier errors when an invalid login is provided

### LINKED ISSUE #29264: MFA required error for SSH to remote node without allowed login
## Expected behavior:
`tsh ssh invalid@node`, where `node` is an SSH node in a remote cluster, should return a helpful access denied error.

## Current behavior:

```
$ ts ssh invalid@node
ERROR: MFA is required to access this resource but user has no MFA devices; use 'tsh mfa add' to register MFA devices
```

The user gets a confusing "MFA is required" error even though MFA is not enabled anywhere in the cluster.

## Bug details:
- Teleport version: 13.x.x
- Recreation steps: attempt to SSH to a node in a remote cluster with a login that is not allowed by any role
- Debug logs

https://github.com/gravitational/teleport/blob/d9ea2061d79b6ceb9484f423c4922836130db9c0/lib/client/api.go#L1649-L1664

### PR DESCRIPTION
**Pull request by rosstimothy**
Any errors from the MFA connection attempt are ignored in favor of the direct connection attempt errors if the target node is in a leaf cluster and MFA is not required to access it.

Fixes #29264.

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -1646,21 +1646,36 @@ func (tc *TeleportClient) ConnectToNode(ctx context.Context, clt *ClusterClient,
...
+		return trace.Wrap(directErr)
 	case !errors.Is(mfaErr, io.EOF) && // Ignore any errors from MFA due to locks being enforced, the direct error will be friendlier
 		!errors.Is(mfaErr, MFARequiredUnknownErr{}) && // Ignore any failures that occurred before determining if MFA was required
 		!errors.Is(mfaErr, services.ErrSessionMFANotRequired): // Ignore any errors caused by attempting the MFA ceremony when MFA will not grant access
-		return nil, trace.Wrap(mfaErr)
+		return trace.Wrap(mfaErr)
 	default:
-		return nil, trace.Wrap(directErr)
+		return trace.Wrap(directErr)
 	}
```
Comment (nklaassen) [inline lib/client/api.go:1679]: I wonder why we can't reduce this to:

```go
if isMFARequiredError(directErr) {
	return trace.Wrap(mfaErr)
}
return trace.Wrap(directErr)
```

Wouldn't it always be this error? It that ever hard to detect for some reason? https://github.com/gravitational/teleport/blob/f30ed729cedac30f8a13d934d507337db0abdaed/lib/services/role.go#L2452

File: lib/client/api.go [hunk 2]
```diff
@@ -1646,21 +1646,36 @@ func (tc *TeleportClient) ConnectToNode(ctx context.Context, clt *ClusterClient,
...
 	switch {
 	// No MFA errors, return any errors from the direct connection
 	case mfaErr == nil:
-		return nil, trace.Wrap(directErr)
+		return trace.Wrap(directErr)
+	// Any access denied errors from a direct connection to a node in another cluster take
+	// precedence over mfa errors. Since the root cluster is always used for MFA we don't
+	// want to show users misleading errors if they don't have any devices in the root cluster.
+	case trace.IsAccessDenied(directErr) && rootCluster != targetCluster:
+		return trace.Wrap(directErr)
```
Comment (nklaassen) [inline lib/client/api.go:1667]: sometimes MFA really is required for remote nodes though, won't this hide the more interesting MFA errors for all remote nodes?

File: lib/client/api.go [hunk 3]
```diff
@@ -1646,21 +1646,36 @@ func (tc *TeleportClient) ConnectToNode(ctx context.Context, clt *ClusterClient,
...
+}
+
+// ChooseConnectionError determines which error from connecting to a host simultaneously
+// with and without MFA should be returned to the user.
+func ChooseConnectionError(directErr, mfaErr error, rootCluster, targetCluster string) error {
 	switch {
 	// No MFA errors, return any errors from the direct connection
 	case mfaErr == nil:
-		return nil, trace.Wrap(directErr)
+		return trace.Wrap(directErr)
```
Comment (nklaassen) [inline lib/client/api.go:1662]: isn't `mfaErr == nil` impossible (we never call this if either path succeeds)

**Review by nklaassen — APPROVED** [review]

File: lib/web/terminal.go [hunk 4]
```diff
@@ -783,7 +805,10 @@ func (t *sshBaseHandler) connectToNode(ctx context.Context, ws WSConn, tc *clien
 		t.sessionData.ServerHostname,
 		tc, modules.GetModules().IsBoringBinary())
 	if err != nil {
-		return nil, trace.NewAggregate(err, conn.Close())
+		// The close error is ignored instead of using [trace.NewAggregate] because
+		// aggregate errors do not allow error inspection with things like [trace.IsAccessDenied].
+		_ = conn.Close()
```
Comment (gabrielcorado) [inline lib/web/terminal.go:810]: nit: any value on logging this with a warning or debug level?

**Review by gabrielcorado — APPROVED** [review]

**Pull request merged**